keps/sig-node/4742-node-labels-downward/README.md

```diff
@@ -0,0 +1,809 @@
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+Exposing node topology labels through the downward API provides a native method
+from which to attain node label values in pods without the need of a sidecar
+to extract information.
+
+## Motivation
+
+We’d like to change the runtime behavior of containers based on node labels.
+In our case, we’re using a CNI with DaemonSets to perform network setup, and
+would like to configure the network differently based on the presence of a node
+label (e.g. `toplogy.k8s.io/vpn=true`).
+
+A number of other use cases exist for providing node labels to pods. One
+example is utilizing topology data from cloud providers, which are automatically
+assignedto nodes, and then used to tailor workloads to minimize cross-zone data
+transfers and reduce costs. Having an easy way for pods to access these node
+topology labels would provide users a straightforward, maintainable way to
+optimize their workloads given topology constraints.
+
+Workarounds today typically involve using an initContainer to query the
+Kubernetes API and then pass data via shared volume to other containers within
+the same pod. This adds additional demand on the API server and is burdensome
+compared to the ease of using downwardAPI for pod labels and metadata.
+
+### Goals
+
+* Gain access to node labels in form of `topology.k8s.io/*` on pods through volume mounts
+* Gain access to node labels in form of `topology.k8s.io/*` on pods through environmental variables
+
+### Non-Goals
+
+* Not to expose additional node info outside of labels
+* Not to pass any additional node labels other than `topology.k8s.io/*` to pods
+* Not to guarantee the label value assigned at pod creation is the most recent node label value because it is assigned at pod creation time
+
+## Proposal
+
+The initial design includes:
+
+* Being able to pass node labels to volumes with fieldPath of `node.metadata.labels` where only labels containing `topology.k8s.io/` are passed through
```

Comment:
fieldpaths are supposed to be paths to fields, not magic words.

What I proposed in https://github.com/kubernetes/kubernetes/issues/40610#issuecomment-2190104439 was to copy (literally copy the data) from node to pod, and define the behavior of that.

Then we don't need a new fieldpath at all.  We just need to decide WHO copies that data, and whether it is literally (and pushed back to apiserver) or virtually copied by kubelet (and kept local, like service link env vars).